sklearn/utils/_param_validation.py

```diff
@@ -410,6 +405,18 @@ def __init__(self, type, left, right, *, closed):
         self._check_params()
 
     def _check_params(self):
+        if self.type not in (Integral, Real, "real_not_int"):
+            raise ValueError(
+                "type must be either numbers.Integral, numbers.Real or 'real_not_int'."
+                f" Got {self.type} instead."
+            )
+
+        if self.closed not in ("left", "right", "both", "neither"):
+            raise ValueError(
+                "closed must be either 'left', 'right', 'both' or 'neither'. "
+                f"Got {self.closed} instead."
+            )
+
```

Comment:
With the removal of `@validate_params`, does `self.left` and `self.right` needs to be validated here?

Reply:
indeed, Added the missing validation